lib/git-fastclone.rb

```diff
@@ -343,43 +337,54 @@ def prefetch(submodule_file)
     # that this repo has been updated on this run of fastclone
     def store_updated_repo(url, mirror, repo_name, fail_hard)
       unless Dir.exist?(mirror)
-        Terrapin::CommandLine.new('git clone', '--mirror :url :mirror')
-                             .run(url: url.to_s, mirror: mirror.to_s)
+        fail_pipe_on_error(
+          ['git', 'clone', verbose ? '--verbose' : '--quiet', '--mirror', url.to_s,
+           mirror.to_s], quiet: !verbose
+        )
       end
 
-      Terrapin::CommandLine.new('cd', ':path; git remote update --prune').run(path: mirror)
-
+      Dir.chdir(mirror) do
+        cmd = ['git', 'remote', verbose ? '--verbose' : nil, 'update', '--prune'].compact
+        if verbose
+          fail_pipe_on_error(cmd, quiet: !verbose)
+        else
+          # Because above operation might spit out a lot to stderr, we use this to swallow them
+          # and only display if the operation return non 0 exit code
+          fail_on_error(*cmd, quiet: !verbose)
+        end
+      end
       reference_updated[repo_name] = true
-    rescue Terrapin::ExitStatusError => e
+    rescue RunnerExecutionRuntimeError => e
       # To avoid corruption of the cache, if we failed to update or check out we remove
       # the cache directory entirely. This may cause the current clone to fail, but if the
       # underlying error from git is transient it will not affect future clones.
-      FileUtils.remove_entry_secure(mirror, force: true)
+      clear_cache(mirror, url)
       raise e if fail_hard
     end
 
     def retriable_error?(error)
       error_strings = [
-        'fatal: missing blob object',
-        'fatal: remote did not send all necessary objects',
-        /fatal: packed object [a-z0-9]+ \(stored in .*?\) is corrupt/,
-        /fatal: pack has \d+ unresolved delta/,
-        'error: unable to read sha1 file of ',
-        'fatal: did not receive expected object',
+        /^fatal: missing blob object/,
+        /^fatal: remote did not send all necessary objects/,
+        /^fatal: packed object [a-z0-9]+ \(stored in .*?\) is corrupt/,
+        /^fatal: pack has \d+ unresolved delta/,
+        /^error: unable to read sha1 file of /,
+        /^fatal: did not receive expected object/,
         /^fatal: unable to read tree [a-z0-9]+\n^warning: Clone succeeded, but checkout failed/
       ]
-      error.to_s =~ /^STDERR:\n.*^#{Regexp.union(error_strings)}/m
+      error.to_s =~ /.*#{Regexp.union(error_strings)}/m
```

Comment:
@justinseanmartin please take a look at this change, i manually throw error and realized it was not captured because the original code assumed the output is from Terrapin
The proposed change here remove that assumption and replace original error_strings with `^` at the beginning so that the union in the end is more generic, but still keeping the assumption that fatal/error msg must start at the beginning of the line)

Reply:
I tested this change via rspec as well as seeing it doing retry if i manually throw error containing one of the error msg above

Reply:
Do we know that those messages appear at the start of the command outputs?

Reply:
Actually, I think this is fine. I see that we are using /m and the full regex is preceded with `.*`. This seems reasonable to me.